### ISSUE TITLE
Wrong TargetFramework: null (regression after fixing the issue #41)

### ISSUE THREAD

**Issue by ycherkes** [ycherkes/ObjectDumper#53]
Nick Westgate: I've used your plugin successfully before (a year or more ago) but today when I tried to use it, it created a new tab "filename.cs" with the following content:  Wrong TargetFramework: null

**Comment by ycherkes** [comment]
**Nick Westgate**,
I need more info from you to reproduce this issue.
Could you please run the following code in your Immediate Window and give me the output?
1. `typeof(System.String).Assembly.Location`
2. `System.AppContext.TargetFrameworkName`

**Comment by ycherkes** [comment]
Related issues: 

- https://github.com/ycherkes/ObjectDumper/issues/41
- https://github.com/ycherkes/ObjectDumper/issues/49

**Comment by sicklittlemonkey** [comment]
Hi. Sure:
```
typeof(System.String).Assembly.Location
"C:\\WINDOWS\\Microsoft.Net\\assembly\\GAC_32\\mscorlib\\v4.0_4.0.0.0__b77a5c561934e089\\mscorlib.dll"
System.AppContext.TargetFrameworkName
".NETFramework,Version=v4.7"

```

**Comment by ycherkes** [comment]
@sicklittlemonkey Thanks, all looks right.

Could you run one more expression: 
`System.AppContext.TargetFrameworkName?.StartsWith(".NETFra") != false ? System.AppContext.TargetFrameworkName : System.AppContext.TargetFrameworkName?.Split(',')[0] + ",Version=v" + System.Environment.Version`

**Comment by sicklittlemonkey** [comment]
Sure:
`".NETFramework,Version=v4.7"`

**Comment by sicklittlemonkey** [comment]
That's the same as what I posted above though.

**Comment by ycherkes** [comment]
Thanks, there are 2 places where error "Wrong TargetFramework: " can appear: https://github.com/ycherkes/ObjectDumper/blob/main/src/ObjectDumper/DebuggeeInteraction/InteractionService.cs#L78-L88

This is interesting because the data you have given me suggests that the error should not occur.
😟

**Comment by ycherkes** [comment]
@sicklittlemonkey
Could you please install the version of ObjectDumper below, try to dump something and then copy the text from the **Output Window -> Show output from: ObjectDumper**? 

![image](https://user-images.githubusercontent.com/13467759/225064461-244a3c62-0d60-437b-a6b9-7ddbfb61374f.png)

[ObjectDumper.zip](https://github.com/ycherkes/ObjectDumper/files/10971165/ObjectDumper.zip)

I've added some tracing, you can see my changes here: https://github.com/ycherkes/ObjectDumper/compare/tracing-for-issue53?expand=1

**Comment by sicklittlemonkey** [comment]
Hi, sure. Sorry for the delay.

It created a new editor window titled objectName.cs with the following content:
`error CS0103: The name 'YellowFlavor' does not exist in the current context`

The ObjectDumper output was:
```
Get expression provider by language: C#
Expression provider successfully found.
Serializer hasn't been injected yet. Trying to inject it.
ObjectDumper package location: c:\users\nickw\appdata\local\microsoft\visualstudio\17.0_99ebe30e\extensions\i24ixs0t.51r
Get serializer directory name by targetFrameworkName: ".NETFramework,Version=v4.7" 
Regex match.Groups["frameworkVersion"].Value: 4.7
Regex match.Groups["frameworkName"].Value: .NETFramework
Version: 4.7
Serializer directoryName: net45 
```

**Comment by ycherkes** [comment]
@sicklittlemonkey, 
No problem, as I see we have a different error now.

Let's try to reproduce all the steps manually in the Immediate window:

1. `System.Reflection.Assembly.LoadFile(@"c:\users\nickw\appdata\local\microsoft\visualstudio\17.0_99ebe30e\extensions\i24ixs0t.51r\InjectableLibs\net45\YellowFlavor.Serialization.dll")
`
2. `System.Text.Encoding.UTF8.GetString(System.Convert.FromBase64String(YellowFlavor.Serialization.ObjectSerializer.Serialize(new[]{1,2,3}, "cs")))`

**Comment by sicklittlemonkey** [comment]
1.
```
{YellowFlavor.Serialization, Version=0.0.0.78, Culture=neutral, PublicKeyToken=null}
    CodeBase: '((System.Reflection.RuntimeAssembly)System.Reflection.Assembly.LoadFile(@"c:\users\nickw\appdata\local\microsoft\visualstudio\17.0_99ebe30e\extensions\i24ixs0t.51r\InjectableLibs\net45\YellowFlavor.Serialization.dll")).CodeBase' threw an exception of type 'System.NotImplementedException'
    CustomAttributes: Method System.Reflection.Assembly.get_CustomAttributes cannot be called in this context.
    DefinedTypes: Method System.Reflection.RuntimeAssembly.get_DefinedTypes cannot be called in this context.
    EntryPoint: null
    EscapedCodeBase: 'System.Reflection.Assembly.LoadFile(@"c:\users\nickw\appdata\local\microsoft\visualstudio\17.0_99ebe30e\extensions\i24ixs0t.51r\InjectableLibs\net45\YellowFlavor.Serialization.dll").EscapedCodeBase' threw an exception of type 'System.NotImplementedException'
    Evidence: '((System.Reflection.RuntimeAssembly)System.Reflection.Assembly.LoadFile(@"c:\users\nickw\appdata\local\microsoft\visualstudio\17.0_99ebe30e\extensions\i24ixs0t.51r\InjectableLibs\net45\YellowFlavor.Serialization.dll")).Evidence' threw an exception of type 'System.NotImplementedException'
    EvidenceNoDemand: Method System.Reflection.RuntimeAssembly.get_EvidenceNoDemand cannot be called in this context.
    ExportedTypes: Method System.Reflection.Assembly.get_ExportedTypes cannot be called in this context.
    Flags: Method System.Reflection.RuntimeAssembly.get_Flags cannot be called in this context.
    FullName: "YellowFlavor.Serialization, Version=0.0.0.78, Culture=neutral, PublicKeyToken=null"
    GlobalAssemblyCache: '((System.Reflection.RuntimeAssembly)System.Reflection.Assembly.LoadFile(@"c:\users\nickw\appdata\local\microsoft\visualstudio\17.0_99ebe30e\extensions\i24ixs0t.51r\InjectableLibs\net45\YellowFlavor.Serialization.dll")).GlobalAssemblyCache' threw an exception of type 'System.NotImplementedException'
    HostContext: '((System.Reflection.RuntimeAssembly)System.Reflection.Assembly.LoadFile(@"c:\users\nickw\appdata\local\microsoft\visualstudio\17.0_99ebe30e\extensions\i24ixs0t.51r\InjectableLibs\net45\YellowFlavor.Serialization.dll")).HostContext' threw an exception of type 'System.NotImplementedException'
    ImageRuntimeVersion: '((System.Reflection.RuntimeAssembly)System.Reflection.Assembly.LoadFile(@"c:\users\nickw\appdata\local\microsoft\visualstudio\17.0_99ebe30e\extensions\i24ixs0t.51r\InjectableLibs\net45\YellowFlavor.Serialization.dll")).ImageRuntimeVersion' threw an exception of type 'System.NotImplementedException'
    InvocableAttributeCtorToken: Method System.Reflection.RuntimeAssembly.get_InvocableAttributeCtorToken cannot be called in this context.
    IsDynamic: Method System.Reflection.RuntimeAssembly.get_IsDynamic cannot be called in this context.
    IsFullyTrusted: Method System.Reflection.Assembly.get_IsFullyTrusted cannot be called in this context.
    IsStrongNameVerified: Method System.Reflection.RuntimeAssembly.get_IsStrongNameVerified cannot be called in this context.
    Location: "c:\\users\\nickw\\appdata\\local\\microsoft\\visualstudio\\17.0_99ebe30e\\extensions\\i24ixs0t.51r\\InjectableLibs\\net45\\YellowFlavor.Serialization.dll"
    ManifestModule: {YellowFlavor.Serialization}
    Modules: Method System.Reflection.Assembly.get_Modules cannot be called in this context.
    PermissionSet: Method System.Reflection.RuntimeAssembly.get_PermissionSet cannot be called in this context.
    ReflectionOnly: false
    SecurityRuleSet: Method System.Reflection.RuntimeAssembly.get_SecurityRuleSet cannot be called in this context.
    SyncRoot: Method System.Reflection.RuntimeAssembly.get_SyncRoot cannot be called in this context.
    _ModuleResolve: Field System.Reflection.RuntimeAssembly._ModuleResolve is not available in this context.
    m_assembly: Field System.Reflection.RuntimeAssembly.m_assembly is not available in this context.
    m_flags: Field System.Reflection.RuntimeAssembly.m_flags is not available in this context.
    m_fullname: Field System.Reflection.RuntimeAssembly.m_fullname is not available in this context.
    m_syncRoot: Field System.Reflection.RuntimeAssembly.m_syncRoot is not available in this context.
```

2.
`error CS0103: The name 'YellowFlavor' does not exist in the current context`

**Comment by ycherkes** [comment]
Wow, seems I was able to reproduce the issue. Are you trying to dump smth in the Release mode?
If so, I mentioned it in the [Know Restrictions](https://github.com/ycherkes/ObjectDumper#known-restrictions):

> the debugging code mustn't be optimized so the Expression Evaluator can be run. If you are trying to debug DLLs from another source, such as a nuget package, you'll get an error message: "Cannot evaluate expression because the code of the current method is optimized". To bypass this you can turn the [Tools > Options > Debugging > General > Suppress JIT optimization on module load](https://learn.microsoft.com/en-us/visualstudio/debugger/jit-optimization-and-debugging?view=vs-2022#the-suppress-jit-optimization-on-module-load-managed-only-option) option on.

Could you switch the configuration to **Debug** or enable **Suppress JIT optimization on module load** and try dump the object again?

**Comment by sicklittlemonkey** [comment]
No, it's debug code, but sorry - for the tests you asked me to do above I was lazy and tried to dump whatever framework object pausing the debugger stopped on, which is probably:
`	private AutoResetEvent _stopSignal = new AutoResetEvent(false);`

For my original problem I was somewhere else in a large codebase that I maintain. I actually can't remember where sorry! I must have breached one of the Known Restrictions, so now I'm conscious of them I will let you know if I hit the problem again.

**Comment by ycherkes** [comment]
OK, let's leave it open for a while.

**Comment by ycherkes** [comment]
Hey @sicklittlemonkey! 
Seems I was able to reproduce the issue in **.net core 2.1** .
Fixed in v 0.0.0.79.
Thanks.
